Gemfile

```diff
@@ -107,7 +107,7 @@ gem 'ratelimit'
 gem 'recaptcha', '4.13.1', require: 'recaptcha/rails'
 gem 'redcarpet', '~>3.5.1', require: false
 gem 'RedCloth', '~>4.3', require: false
-gem 'redis', '~> 4.2.0', require: ['redis', 'redis/connection/hiredis']
+gem 'redis', require: ['redis']
```

Comment:
Sorry, I actually misread it :grimacing:  I thought this was for the new `redis-client` gem...

So, we have two gems: `redis` for non-sidekiq stuff and `redis-client` for sidekiq stuff, right?
But for `redis` we dropped `hiredis`, why is that? It is not compatible?

Reply:
All the Redis gems ecosystem is highly confusing, since all gems have basically the same name 😵‍💫. I'll try to summarize it:

- First we have the `redis` gem: https://github.com/redis/redis-rb
  - It provides a Ruby API for the Redis protocol, that is, ruby methods like `:sadd`, `:hset`, etc.
  - Also provides thread safe connections with connections pools.
- `redis` needs a client, since 5.0 they use `redis-client`: https://github.com/redis-rb/redis-client
  - It provides the connection to server but it's not thread safe and doesn't implement the redis protocol, only provides a `call` method to be used like this: `client.call("SADD", "key", "value")`.
- `redis-client` requires a redis driver. It can work with a builtin Ruby driver, which is slow, or with `hiredis`, which is faster. `hiredis-client` is the gem that provides `hiredis` for `redis-client`. https://github.com/redis-rb/redis-client/tree/master/hiredis-client
- `hiredis-client` is just a ruby wrap for the C `hiredis` library: https://github.com/redis/hiredis
- On the other hand, there's a ruby gem called `hiredis`: https://github.com/redis/hiredis-rb
  - It's also a wrapper for the C library. However, this one is outdated and doesn't support TLS connections.

So before updating `redis-rb` we had:
- `redis` (4.X) -> builtin client -> `hiredis` (ruby) -> `hiredis` (C)

And now:
- `redis` (5.X) -> `redis-client` -> `hiredis-client` -> `hiredis` (C)

> So, we have two gems: redis for non-sidekiq stuff and redis-client for sidekiq stuff, right?

Kind of, porta uses `redis` API (e.g. [/app/models/billing_summary.rb](https://github.com/3scale/porta/blob/3scale-2.13.2-GA/app/models/billing_summary.rb)), but `redis` depends on `redis-client`, so they aren't two different gems to do the same. Sidekiq calls `redis-client` directly (See [here](https://github.com/sidekiq/sidekiq/blob/v7.1.6/lib/sidekiq/redis_client_adapter.rb#L26-L32))

> But for redis we dropped hiredis, why is that? It is not compatible?

Because it doesn't support TLS (https://github.com/redis/hiredis-rb/issues/58)

> I'm wondering, what if we use redis-client for non-sidekiq also?

Porta uses `redis` API, so we would need to invent some solution like Sidekiq does. Apisonator also has something similar (https://github.com/3scale/apisonator/blob/v3.4.3/lib/3scale/backend/storage_async/client.rb)

Reply:
OMG, this is extremely confusing indeed :face_with_spiral_eyes: 
Thanks a lot for this comment, it clarifies a lot!